### PR TITLE
ZIP 212 NCC fixes

### PR DESCRIPTION
**Pull request by therealyingtong**
Fixes to the ZIP 212 implementation as suggested in the NCC audit.

### CODE REVIEW

File: src/zcash/prf.h [hunk 1]
```diff
@@ -25,6 +25,12 @@ uint256 PRF_ovk(const uint256& sk);
 uint256 PRF_rcm(const uint256& rseed);
 uint256 PRF_esk(const uint256& rseed);
 
+const char PRF_ASK_DIVERSIFIER = 0;
+const char PRF_NSK_DIVERSIFIER = 1;
+const char PRF_OVK_DIVERSIFIER = 2;
+const char PRF_RCM_DIVERSIFIER = 4;
+const char PRF_ESK_DIVERSIFIER = 5;
```
Comment (str4d) [inline src/zcash/prf.h:32]: "Diversifier" conflates these with diversifiers in the Sapling protocol. The protocol spec uses the symbol `t` for these, which I think means "tag", so:
```suggestion
const char PRF_ASK_TAG = 0;
const char PRF_NSK_TAG = 1;
const char PRF_OVK_TAG = 2;
const char PRF_RCM_TAG = 4;
const char PRF_ESK_TAG = 5;
```

**Review by str4d — CHANGES_REQUESTED** [review]
utACK after renaming.

**Review by str4d — APPROVED** [review]
utACK

**Comment by nuttycom** [comment]
utACK

**Comment by therealyingtong** [comment]
@zkbot r+

**Pull request merged**